cstar/base/component.py

```diff
@@ -158,3 +138,66 @@ def post_run(self) -> None:
         This abstract method will be implemented differently by different Component types.
         """
         pass
+
+
+class Discretization:
+    """
+    Holds discretization information about a Component.
+
+    Attributes:
+    -----------
+
+    time_step: int
+        The time step with which to run ROMS in this configuration
+    nx,ny,n_levels: int
+        The number of x and y points and vertical levels in the domain associated with this object
+    """
+
+    def __init__(
+        self,
+        time_step: int,
+        nx: Optional[int] = None,
+        ny: Optional[int] = None,
+        n_levels: Optional[int] = None,
+    ):
+        """
+        Initialize a Discretization object from basic discretization parameters
+
+        Parameters:
+        -----------
+        time_step: int
+            The time step with which to run the Component
+        nx,ny,n_levels: int
+            The number of x and y points and vertical levels in the domain associated with this object
+
+        Returns:
+        --------
+        Discretization:
+            An initialized Discretization object
+
+        """
+
+        self.time_step: int = time_step
+        self.nx: Optional[int] = nx
+        self.ny: Optional[int] = ny
+        self.n_levels: Optional[int] = n_levels
```

Comment:
Surely this information is redundant with the `Grid` object?

Reply:
It actually wasn't being used anyway so I've just taken it out in b69800f70ca3a583ffdde906d663d847df97404b